### PR TITLE
add configuring dashboard service type to helm chart

### PR DESCRIPTION
**Pull request by thojkooi**
Signed-off-by: Thomas Kooi <t.j.kooi@avisi.nl>

### What problem does this PR solve?

Allow creating a service for the Dashboard without exposing it as a nodePort in the helm chart.

### What is changed and how does it work?

Make `type`, `clusterIP` and `nodePort` fields of the dashboard service configurable, while defaulting to the current defaults (`NodePort`).

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
the integration test failed because of the file  of crd is not updated in remote, just ignore it now

**Comment by zhouqiang-cl** [comment]
@YangKeao @cwen0 @fewdan PTAL

File: helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml [hunk 1]
```diff
@@ -92,10 +92,16 @@ spec:
     app.kubernetes.io/name: {{ template "chaos-mesh.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/component: chaos-dashboard
-  type: NodePort
+  type: {{ .Values.dashboard.service.type }}
+{{- if (and (eq .Values.dashboard.service.type "ClusterIP") (not (empty .Values.dashboard.service.clusterIP))) }}
```
Comment (Gallardot) [inline helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml:96]:
```suggestion
{{- if and .Values.dashboard.service.clusterIP (eq .Values.dashboard.service.type "ClusterIP") }}
```

File: helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml [hunk 2]
```diff
@@ -92,10 +92,16 @@ spec:
...
+  clusterIP: {{ .Values.dashboard.service.clusterIP }}
+{{- end }}
   ports:
     - protocol: TCP
       port: {{ .Values.dashboard.env.LISTEN_PORT }}
       targetPort: {{ .Values.dashboard.env.LISTEN_PORT }}
       name: http
+{{- if (and (eq .Values.dashboard.service.type "NodePort") (not (empty .Values.dashboard.service.nodePort))) }}
```
Comment (Gallardot) [inline helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml:104]:
```suggestion
{{- if and .Values.dashboard.service.nodePort (eq .Values.dashboard.service.type "NodePort") }}
```

**Review by Gallardot — COMMENTED** [review]
rest LGTM

File: helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml (hunk 2 above)
Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-dashboard-deployment.yaml:104]: @thojkooi PTAL, Thanks!

**Comment by zhouqiang-cl** [comment]
@thojkooi Please take a look at the comments, thank you very much

**Comment by thojkooi** [comment]
> @thojkooi Please take a look at the comments, thank you very much

Sorry, I've been away for a bit.
I've made the suggested changes and rebased the branch.

**Review by TobiColada — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]
LGTM

**Review by STRRL — COMMENTED** [review]
Hi @thojkooi , how about spliting Service to another file like "chaos-dashboard-service.yaml"?

**Comment by YangKeao** [comment]
@STRRL It can be done in another PR.

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**